sw/host/opentitanlib/src/transport/verilator/subprocess.rs

```diff
@@ -94,3 +94,35 @@ impl Subprocess {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    fn echo_subprocess() -> Result<Subprocess> {
+        let options = Options {
+            executable: "/bin/echo".to_owned(),
+            rom_image: "".to_owned(),
+            flash_image: "".to_owned(),
+            otp_image: "".to_owned(),
+            extra_args: vec!["abc 123 def 456".to_owned()],
+        };
+        Subprocess::from_options(options)
+    }
+
+    #[test]
+    fn test_find_regex() -> Result<()> {
+        let mut subprocess = echo_subprocess()?;
+        let regex = Regex::new("abc (.*) def")?;
+        let found = subprocess.find(&regex, Duration::from_millis(100))?;
+        assert_eq!(found, "123");
+        Ok(())
+    }
+
+    #[test]
+    fn test_kill() -> Result<()> {
+        let mut subprocess = echo_subprocess()?;
+        subprocess.kill()?;
```

Comment:
Is there a race condition here? `kill` returns `InvalidInput` if the child has already exited.

Reply:
Good catch.  I've added some logic to the kill function to detect this condition.  I can't seem to provoke that error in the test though - even sleeping for 5s after the `echo` command.  I suspect that for `kill`, "already exited" might mean "already exited and has been reaped such that the process is no longer around, not even in the zombie state".